addons/hr_expense/models/hr_expense.py

```diff
@@ -1141,7 +1141,13 @@ def _check_can_approve(self):
 
     def approve_expense_sheets(self):
         self._check_can_approve()
-
+        if self.env.context.get('validate_analytic', False):
```

Comment:
AML has a `_validate_distribution` method, but the others (expense, SO, PO) don't

Reply:
Indeed, they directly use the validate_analytic_distribution from the mixin. 
I don't know if we want a validate_distribution function everywhere (I would personally remove the function from the AML, but if I'm not mistaken, removing a function or changing its declaration is bad stable policy, so I would do that in master).

Reply:
ah bummer! it's in the `_create_analytic_lines` method! oh well.